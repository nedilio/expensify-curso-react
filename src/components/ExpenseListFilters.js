import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from "../actions/filters";

class ExpenseListFilters extends React.Component {
  state = {
    calendarFocused: null,
  };

  onDatesChange = ( {startDate,endDate}) => {
    this.props.dispatch(setStartDate(startDate));
    this.props.dispatch(setEndDate(endDate));
  };

  onFocusChange = (calendarFocused) => {
    this.setState( () => ({ calendarFocused }))
  };

  render(){
    return (
      <div>
        <input 
          type="text" 
          defaultValue={this.props.filters.text} onChange={(e) => {
            this.props.dispatch(setTextFilter(e.target.value));
          }}
        />
        <select 
          defaultValue={this.props.filters.sortBy} 
          onChange={(e)=>{
            const sortBy = e.currentTarget.value;
            if (sortBy === 'date') {
              this.props.dispatch(sortByDate());
            } else {
              this.props.dispatch(sortByAmount());
            }
          }}
        >
          <option value="amount">Amount</option>
          <option value="date">Date</option>
        </select>
        <DateRangePicker 
          startDate={this.props.filters.startDate}
          endDate={this.props.filters.endDate}
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.calendarFocused}
          onFocusChange={this.onFocusChange}
          numberOfMonths={1}
          isOutsideRange={() => false}
          showClearDates={true}
        />
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  }
}

export default connect(mapStateToProps)(ExpenseListFilters);