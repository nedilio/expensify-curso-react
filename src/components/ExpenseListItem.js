import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

numeral.register('locale', 'cl', {
  delimiters: {
      thousands: '.',
      decimal: ','
  },
  abbreviations: {
      thousand: 'k',
      million: 'm',
      billion: 'b',
      trillion: 't'
  },
  currency: {
      symbol: '$'
  }
});

// switch between locales
numeral.locale('cl');

const ExpenseListItem = ({ description, amount, createdAt, id}) => (
  <div>
    <Link to={`/edit/${id}`}>
    <h3>{description}</h3></Link>
    <p>{numeral(amount).format('$ 0.0[,]00')}
    - 
    {moment(createdAt).format('D MMMM, YYYY')}</p>
  </div>
);

export default ExpenseListItem;