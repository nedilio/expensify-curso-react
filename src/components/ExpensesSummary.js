import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

 export const ExpensesSummary = ({ expenseCount, expensesTotal}) => {
  const expenseWord = expenseCount === 1 ? 'gasto' : 'gastos';
  const formatExpensesTotal = numeral(expensesTotal).format('$ 0.0[,]00');
  return (
    <div>
      <h1>Mostrando {expenseCount} {expenseWord} totalizando {formatExpensesTotal}</h1>
    </div>
  );
};

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);