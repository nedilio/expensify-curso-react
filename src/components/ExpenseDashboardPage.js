import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';

const ExpenseDashboardPage = () => (
  <div>
    <p>
      This is from my Dashboard Component
    </p>
    <ExpenseListFilters />
    <ExpenseList />
  </div>
);

export default ExpenseDashboardPage;
