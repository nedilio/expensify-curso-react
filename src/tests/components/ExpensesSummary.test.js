import React from 'react'
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test ('render expeneses summary with 1 expense', () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={50000}/>);
  expect(wrapper).toMatchSnapshot();
});

test ('render expeneses summary with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={55} expensesTotal={8234857328945}/>);
  expect(wrapper).toMatchSnapshot();
});