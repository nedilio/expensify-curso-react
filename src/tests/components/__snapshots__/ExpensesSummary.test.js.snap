// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render expeneses summary with 1 expense 1`] = `
<div>
  <h1>
    Mostrando 
     
    gastos
     totalizando 
    $ 50,000.0
  </h1>
</div>
`;

exports[`render expeneses summary with multiple expenses 1`] = `
<div>
  <h1>
    Mostrando 
     
    gastos
     totalizando 
    $ 8,234,857,328,945.0
  </h1>
</div>
`;
